### PR TITLE
Using less spilling values for kOuter

### PR DESCRIPTION
**Pull request by giuseros**
This is trying to address #1245 . Immediate changes
- I put the last config parameter back to `true` 
- I removed all the `kouter={16/32}` configurations from both the tuning and the default list. They spill in most of the cases, and if they don't spill they are outperformed by configuration with smaller `kOuter`. 
- I added the possibility to have `kPack < kBase(16)` .While for xdlops we had kbase < maxVectorLength, this is not the case for wmma. Indeed, kpack=16 would vectorize properly only for int8. 

Long term plan: find a new set of "default" configurations

### CODE REVIEW

**Comment by giuseros** [comment]
@krzysz00 maybe you could try to see if this fixes the hangs you are experiencing

File: mlir/lib/Dialect/Rock/Transforms/AccelEmitter.h [hunk 1]
```diff
@@ -110,6 +110,9 @@ struct AccelEmitter {
                                        Location loc, Value baseOffset,
                                        Value dWaves, Value laneId) = 0;
 
+  /// Validate the accelerator structure
+  virtual void validateAcceleratorProperties(){};
```
Comment (krzysz00) [inline mlir/lib/Dialect/Rock/Transforms/AccelEmitter.h]: `virtual LogicalResult` and then crashing on failure or something.

File: mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp [hunk 2]
```diff
@@ -710,50 +710,18 @@ const InitParamsAccel
...
-  {32, 64, 16, 32, 32, 16, true, true},
-  {32, 64, 16, 16, 64, 16, true, true},
-  {32, 64, 16, 16, 32, 16, true, true},
   {32, 64, 4, 32, 64, 16, true, true},
-  {32, 32, 32, 16, 32, 16, true, true},
-  {32, 32, 16, 32, 32, 16, true, true},
-  {32, 32, 16, 16, 16, 16, true, true},
-  {32, 32, 16, 16, 32, 16, true, true},
-  {16, 64, 16, 16, 64, 16, true, true},
-  {16, 64, 16, 16, 32, 16, true, true},
```
Comment (krzysz00) [inline mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp:727]: Let's put these back as `kPerBlock = 4` or, more generally, re-run.

File: mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp [hunk 3]
```diff
@@ -57,10 +57,10 @@ void createGemmTuningRangeBF(TuningParamSet *newSpace,
   const std::vector<std::vector<uint32_t>> validRangeWmmaGemmParams = {
       {4, 8, 16, 32, 64, 128, 256},
       {16, 32, 64, 128, 256},
-      {16, 32},
+      {1, 2, 4, 8},
       {4, 8, 16, 32, 64, 128},
       {4, 8, 16, 32, 64, 128},
-      {16},
+      {4, 8, 16},
```
Comment (krzysz00) [inline mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp:63]: Will we want kPack < 16?

(I think so but)

Comment (giuseros) [reply mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp:63]: Yes, for two reasons:
- For fp16 the maximum hardware vector is 8 (128bits)
- In general we saw that there are configurations where a vector smaller than the max allowed  is faster

File: mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp (hunk 2 above)
Comment (giuseros) [reply mlir/lib/Dialect/Rock/Tuning/GridwiseGemmParams.cpp:727]: Coming up in a follow-up PR

**Review by krzysz00 — APPROVED** [review]

**Comment by giuseros** [comment]
I stopped at analyzing ~240 gemm configs, but the results look good enough: 
- "Quick tuning" time reduced by a lot on navi32 (I didn't measure, but it was more than a 2x reduction)
- Performance improved by a lot
- There are 4 cases where performance decreased, but only 1 is real (and it is about 10% hit on a very small int8 gemm)

I am attaching the report here
[navi32.zip](https://github.com/ROCmSoftwarePlatform/rocMLIR/files/12699829/navi32.zip)

**Review by krzysz00 — APPROVED** [review]

**Comment by giuseros** [comment]
Given the performance improvements and the simplicity of the change, I am merging this in. If worse comes to worst, we can always revert this change using the old configs.

**Pull request merged**